index-dot-shaka-player-demo.appspot.com broken

<!-- NOTE: If you ignore this template, we will send it again and ask you to fill it out anyway. -->

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?** Yes

**What did you do?**
Open https://index-dot-shaka-player-demo.appspot.com/


**What did you expect to happen?**
v3 versions are shown 

**What actually happened?**
v3 versions are missing

The call to `google.appengine.api.modules.modules.get_versions()` is ultimately failing with `AssertionError: No api proxy found for service "modules"`.  This worked fine with the now-deprecated Python 2 engine, and it's not clear that there's a functioning equivalent in the new runtime.  (Or maybe there is, but it requires the setup of additional services or something.)

Since the Google Cloud docs are unhelpful on this, I suggest we just stop depending on this app-engine-specific code at all.  Back when the index service was launched, it was its own thing, and looking up appspot subdomains from inside appspot made sense.

Now that we have GitHub Actions, there's no reason for any dynamic code in this thing at all.  It could be purely static content, regenerated and redeployed each time a release is tagged on GitHub.

I think that's the way forward.  It's also independent of Google Cloud.  Even if we have no plans to host it elsewhere, the index will be more portable this way.

--- a/.github/workflows/demo-version-index.yaml
+++ b/.github/workflows/demo-version-index.yaml
@@ -17,6 +17,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        # We need a list of all tags for this, so fetch the entire history.
+        fetch-depth: 0
 
       - name: Generate static content
         run: python3 app-engine/demo-version-index/generate.py

